### ISSUE TITLE
When on typescript project, sequelize-cli still generates javascript files.

### ISSUE THREAD

**Issue by malik-samad** [sequelize/cli#1436]
## What I am doing?
I'm using Sequelize-cli in a typescript project and it is not generating .ts files and I have to then update those files extensions and make manual changes to the file for making it compatible in my project.

## What I expect to happen?
I want to have the CLI work the same for my typescript file as it does for a javascript project. I expect the cli to generate proper migration, see and model files with .ts extension.

## What is actually happening?
When I try to run `sequelize init` or other migration cli, it will generate javascript files instead of typescript as expected.

Dialect: mysql
Database version: latest
Sequelize CLI version: latest
Sequelize version: latest

**Comment by WikiRik** [comment]
Thanks for opening this! You can fork this repository, push the changes to there and then open a pull request here

**Comment by berubenic** [comment]
+1

**Comment by malik-samad** [comment]
Hi WikiRik, 
I have opened a PR for this issue and it needs your review. please have a look and help me get in making it merged into package.

The PR has 1 regex change which is reported by CodeQL and it needs approving from reviewer as I think its a false positive.

thanks.

**Comment by IgorArnaut** [comment]
I wish there was an option to generate TypeScript version of the project.

@malik-samad
A year later and your pull request is still not merged.

**Comment by omertairi04** [comment]
Why exactly isn't this merged ?

**Comment by malik-samad** [comment]
> Why exactly isn't this merged ?

Hi, thanks for checking.
The reason is because the PR needs approval from the reviewer. I can't merge it on my own.

**Comment by omertairi04** [comment]
> > Why exactly isn't this merged ?
> 
> Hi, thanks for checking. The reason is because the PR needs approval from the reviewer. I can't merge it on my own.

Since i am a beginner when it comes to things as such, is there any work around to implement your solution ?

**Comment by shlomisas** [comment]
@WikiRik any update on ETA to approve the PR? 🙏

**Comment by IgorArnaut** [comment]
> [@WikiRik](https://github.com/WikiRik) any update on ETA to approve the PR? 🙏
There's no ETA. The main devs just don't care.

**Comment by malik-samad** [comment]
> > > Why exactly isn't this merged ?
> > 
> > 
> > Hi, thanks for checking. The reason is because the PR needs approval from the reviewer. I can't merge it on my own.
> 
> Since i am a beginner when it comes to things as such, is there any work around to implement your solution ?

Yes, You can actually check my other repo for this implementation.

https://github.com/malik-samad/sequelize-with-typscript

the repo link above contains the initial boilerplate implementation of using typescript with Sequelize. in that repo you will that I have created sequelize models in typescript instead of javascript. and it is working just fine. 

(Until my PR is not merged in the Sequelize-CLI repo) you can also use my modified Sequelize-cli package to make it support the typescript
NPM: https://www.npmjs.com/package/sequelize-cli-ts-js
Repo: https://github.com/malik-samad/sequelize-cli-ts-js

I hope this will help in your situation.

**Comment by malik-samad** [comment]
> [@WikiRik](https://github.com/WikiRik) any update on ETA to approve the PR? 🙏

check my repos and NPM package for your reference to get you blocker resolved until my PR gets approved and merged.

I don't know how long it may take, maybe it never get merged 😄 seems like the repo is abandoned or something. 

the workaround I did is tested and used in production level projects.

**Comment by malik-samad** [comment]
> > [@WikiRik](https://github.com/WikiRik) any update on ETA to approve the PR? 🙏
> 
> There's no ETA. The main devs just don't care.

yeh, you are right. otherwise I should atleast get some response by now. its has been more than a year now since I have opened  the PR.
